AdyenNetworkingTests/APIClientTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  APIClientTests.swift
+//  AdyenNetworkingTests
+//
+//  Created by Alexander Guretzki on 07/02/2025.
+//
+
+import Testing
+import Foundation
+@testable import AdyenNetworking
+
+@MainActor
+struct APIClientTests {
+    
+    struct TestResponse {
+        let data: Data?
+        let response: HTTPURLResponse?
+        let error: Error?
+
+        init(
+            data: Data? = nil,
+            statusCode: Int? = nil,
+            error: Error? = nil
+        ) {
+            self.data = data
+            self.response = statusCode.map { .with(statusCode: $0) }
+            self.error = error
+        }
+    }
+    
+    @Test func emptyResponse() async throws {
```

Comment:
Dies Swift Testing suggests a naming convention? Anyhow I'd prefer to read a test name and move on with no need to understand HOW tests works. Main idea is to generate mind map for sut coverage by just reading test names.

Suggestion: `@Test func client_fails_onEmptyResponse() async throws {`

Reply:
There is no convention - they removed the requirement to start the names with `test` so I only removed that part - can double check on the naming - what do you think of the change?

Reply:
reads much better 👍